spec/models/redirect_spec.rb

```diff
@@ -0,0 +1,32 @@
+describe Redirect do
+  describe 'validations' do
+    subject do
+      described_class.new(source: 'foo',
+                          destination: 'bar',
+                          redirect_type: 'temporary')
+    end
+
+    it { is_expected.to validate_presence_of(:source)  }
+    it { is_expected.to validate_presence_of(:destination)  }
+    it { is_expected.to validate_presence_of(:redirect_type)  }
+    it { is_expected.to validate_uniqueness_of(:source) }
+
+    it 'validates source and destination not identical' do
+      subject.destination = 'foo'
+      expect(subject).to_not be_valid
```

Comment:
I've changed the error from base to destination attribute as I think that its better.

Regarding the specific error I think the test description is succinct enough and will be displayed should the test fail.